### ISSUE TITLE
Support generating a Patch file instead of applying changes.

### ISSUE THREAD

**Issue by JoeRobich** [dotnet/format#1177]
Maybe as a different reporting format we could generate a patch file.

**Comment by RikkiGibson** [comment]
related to #789?

**Comment by JoeRobich** [comment]
> related to #789?

@RikkiGibson It was a request from @tannergooding way back when we first released.

**Comment by tannergooding** [comment]
Yep, it would be super usefal to generate a `format.patch` file that you can apply to a repo.

The runtime repo, for example, has this generated as part of the C++ format validation job powered by `clang-format` and this makes it really easy to download and apply.
